includes/writer.js

```diff
@@ -32,15 +32,17 @@ module.exports = function(db, bufferSize, parallelism, ee) {
     // if we are restoring known revisions, we need to supply new_edits=false
     if (payload.docs && payload.docs[0] && payload.docs[0]._rev) {
       payload.new_edits = false;
+      debug('Using new_edits false mode.');
     }
 
     if (!didError) {
       db.service.postBulkDocs({
         db: db.db,
         bulkDocs: payload
       }).then(response => {
-        if (!response.result || response.result.length > 0) {
-          throw new Error('Error writing batch.');
+        if (!response.result || (payload.new_edits === false && response.result.length > 0)) {
```

Comment:
IMO a test case when payload does not have `new_edits` would fail here. Can this happen in practice?
E.g. the payload arrives without revs AND without the `new_edits` param?

You wrote in the issue description:
> It is worth noting that all backups produced after `2.4.2` include `_rev` info, but shallow backups created prior to `2.4.2` did not include `_rev` info.

Reply:
>a test case when payload does not have new_edits would fail here

fail the `if` check, or return the error?

>Can this happen in practice?
>E.g. the payload arrives without revs AND without the new_edits param?

yes, this is exactly what happens in a restore of a <2.4.2 shallow backup

The setting of the `new_edits` property is governed by the presence of `_rev` in the doc content of the backup (it's [line 33/34](https://github.com/cloudant/couchbackup/pull/384/files/1edf6cf76222d39af34a8ecc7227810f706b723b#diff-3b76bddc4e460cc434375132ce2fbdd645bc47451c2d93512bb00df275e60296R34))

This means that for backups without `_rev` info, `new_edits` is `undefined` and the restore has `new_edits=true` behaviour on the request.

That is the reason I used the check `payload.new_edits === false` and not `payload.new_edits` because the `undefined` `new_edits` state is _falsy_, but does not correspond to `new_edits=false` behaviour and in the `new_edits=true` (i.e. `undefined`) case we don't want to error if the array has content.

This is the behaviour I wanted:
| `payload.new_edits` | `new_edits` on request | should check array length=0 |
| --- | --- | --- |
| `false` | `false` | yes, expect empty array |
| `true` | `true` | no, expect array content (note we don't explcitly set `true`) |
| `undefined` | `true` | no, expect array content |

Do you think the code delivers that? Did I miss something?

Reply:
> fail the `if` check, or return the error?

Meanwhile, I realized Node has the type check (triple =s). I think I spent too much time in a Python environment lately huh? So basically if `payload.new_edits` is `undefined`, then `payload.new_edits === false` will be false, gotcha.

Reply:
np, thanks for being thorough better to double-check now than break later.